index.js

```diff
@@ -1026,26 +1026,32 @@ class protractorImageComparison {
      * @param {object} options (non-default) options
      * @param {int} options.resizeDimensions the value to increase the size of the element that needs to be saved
      * @param {boolean} options.disableCSSAnimation enable or disable CSS animation
+     * @param {boolean} options.canvasScreenshot enable or disable taking screenshot directly from canvas (via dataUrl instead of browser.takeScreenshot())
      * @returns {Promise} The images has been saved when the promise is resolved
      * @public
      */
-    saveElement(element, tag, options) {
+    async saveElement(element, tag, options) {
         let saveOptions = options || [];
         let bufferedScreenshot;
+        let screenshot;
 
         this.saveType.element = true;
         this.resizeDimensions = saveOptions.resizeDimensions ? saveOptions.resizeDimensions : this.resizeDimensions;
         this.disableCSSAnimation = saveOptions.disableCSSAnimation || saveOptions.disableCSSAnimation === false ? saveOptions.disableCSSAnimation : this.disableCSSAnimation;
 
-        return this._getInstanceData()
-            .then(() => browser.takeScreenshot())
-            .then(screenshot => {
-                bufferedScreenshot = new Buffer(screenshot, 'base64');
-                this.screenshotHeight = (bufferedScreenshot.readUInt32BE(20) / this.devicePixelRatio); // width = 16
+        await this._getInstanceData();
+        if(saveOptions.canvasScreenshot) {
+            const dataUrl = await browser.executeScript((canvas) => canvas.toDataURL('image/png'), await element.getWebElement());
```

Comment:
This is new to me, tnx for this, but what is the support of `canvas`, do all (mobile) browsers support this? If not we need to make this or available on own risk, or based on a browser version

Reply:
This works only for <canvas> elements, which implements toDataURL function. More information: https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL. There is compability chart which says it does not work in Edge mobile browser, but for others it should be OK.

Reply:
Ok, tnx, I'll dive into it, but for now I'm also missing:

- documentation, due to the boolean you can't add `resizeDimensions`
- tests
- the usage of `async/await` needs to be removed. 

If you add that I can add it as a nice feature